client/accounting.go

```diff
@@ -7,76 +7,122 @@ import (
 	v2accounting "github.com/nspcc-dev/neofs-api-go/v2/accounting"
 	rpcapi "github.com/nspcc-dev/neofs-api-go/v2/rpc"
 	"github.com/nspcc-dev/neofs-api-go/v2/rpc/client"
-	v2signature "github.com/nspcc-dev/neofs-api-go/v2/signature"
 	"github.com/nspcc-dev/neofs-sdk-go/accounting"
 	"github.com/nspcc-dev/neofs-sdk-go/owner"
 )
 
 // Accounting contains methods related to balance querying.
 type Accounting interface {
 	// GetBalance returns balance of provided account.
-	GetBalance(context.Context, *owner.ID, ...CallOption) (*BalanceOfRes, error)
+	GetBalance(context.Context, GetBalancePrm) (*GetBalanceRes, error)
 }
 
-type BalanceOfRes struct {
+// GetBalancePrm groups parameters of GetBalance operation.
+type GetBalancePrm struct {
+	ownerSet bool
+	ownerID  owner.ID
+}
+
+// checks if all parameters are set correctly and throws panic if not.
+func (x GetBalancePrm) check() {
+	switch {
+	case !x.ownerSet:
+		panic("account not set")
+	case !x.ownerID.Valid():
+		panic("invalid account ID")
+	}
+}
+
+// SetAccount sets identifier of the NeoFS account for which the balance is requested.
+// Required parameter. Must be a valid ID according to NeoFS API protocol.
+func (x *GetBalancePrm) SetAccount(id owner.ID) {
+	x.ownerID = id
+	x.ownerSet = true
+}
+
+// GetBalanceRes groups resulting values of GetBalance operation.
+type GetBalanceRes struct {
 	statusRes
 
 	amount *accounting.Decimal
 }
 
-func (x *BalanceOfRes) setAmount(v *accounting.Decimal) {
+func (x *GetBalanceRes) setAmount(v *accounting.Decimal) {
 	x.amount = v
 }
 
-func (x BalanceOfRes) Amount() *accounting.Decimal {
+// Amount returns current amount of funds on the NeoFS account as decimal number.
+//
+// Client doesn't retain value so modification is safe.
+func (x GetBalanceRes) Amount() *accounting.Decimal {
 	return x.amount
 }
 
-func (c *clientImpl) GetBalance(ctx context.Context, owner *owner.ID, opts ...CallOption) (*BalanceOfRes, error) {
-	// apply all available options
-	callOptions := c.defaultCallOptions()
-
-	for i := range opts {
-		opts[i](callOptions)
+// GetBalance requests current balance of the NeoFS account.
+//
+// Immediately panics if parameters are set incorrectly (see GetBalancePrm docs).
+// Context is required and must not be nil. It is used for network communication.
+//
+// Exactly one return value is non-nil. Server status return is returned in GetBalanceRes.
+// Reflects all internal errors in second return value (transport problems, response processing, etc.).
+func (c *clientImpl) GetBalance(ctx context.Context, prm GetBalancePrm) (*GetBalanceRes, error) {
+	// check context
+	if ctx == nil {
```

Comment:
Won't it panic anyway if `ctx == nil`?

Reply:
It will, but much later and caused by NPE which is not clear enough.